samples/boards/esp32/wifi_station/src/main.c

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2020 Espressif Systems (Shanghai) Co., Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <sys/printk.h>
+#include <esp_wifi.h>
+#include <esp_timer.h>
+#include <esp_event.h>
+
+#include <net/net_if.h>
+#include <net/net_core.h>
+#include <net/net_context.h>
+#include <net/net_mgmt.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(esp32_wifi_sta, LOG_LEVEL_DBG);
+
+static struct net_mgmt_event_callback mgmt_cb;
+
+static void handler(struct net_mgmt_event_callback *cb,
+		    uint32_t mgmt_event,
+		    struct net_if *iface)
+{
+	int i = 0;
+
+	if (mgmt_event != NET_EVENT_IPV4_ADDR_ADD) {
+		return;
+	}
+
+	for (i = 0; i < NET_IF_MAX_IPV4_ADDR; i++) {
+		char buf[NET_IPV4_ADDR_LEN];
+
+		if (iface->config.ip.ipv4->unicast[i].addr_type !=
+							NET_ADDR_DHCP) {
+			continue;
+		}
```

Comment:
You can simplify this as the DHCPv4 module sends BOUND events atm. This support was added recently.
See usage example in `tests/net/dhcpv4/src/main.c`

Reply:
@jukkar, not sure what could be simplified in here unless replacing `NET_EVENT_IPV4_ADDR_ADD`  to `NET_EVENT_IPV4_DHCP_BOUND`. It only prints interface IP address, lease time, etc..

Reply:
@sylvioalves sure, I was just saying that the event handler can get the printed information directly from the event info, instead of looping through addresses that the interface has.